test/unittests/bit_cast_test.cpp

```diff
@@ -0,0 +1,21 @@
+// Fizzy: A fast WebAssembly interpreter
+// Copyright 2020 The Fizzy Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "cxx20/bit.hpp"
+#include <gtest/gtest.h>
+
+using namespace fizzy;
+using namespace testing;
+
+// Test cases from https://en.cppreference.com/w/cpp/numeric/bit_cast#Example.
```

Comment:
Is it supposed to work between different signs, e.g uint64<->int64? If so, can you add an example? If not can you add a `static_assert` in the impl?

Reply:
Yes, it should work for types of different signness. Also for arrays, structs and other C-like types.